### ISSUE TITLE
BboxConnectorPatch does not show facecolor

### ISSUE THREAD

**Issue by jotasi** [matplotlib/matplotlib#8059]
### Bug report

**Bug summary**

When adding a BboxConnectorPatch, the edges are displayed correctly but the facecolor is not shown. The face is empty.

**Code for reproduction**
```python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.transforms import Bbox, TransformedBbox, \
     blended_transform_factory
from mpl_toolkits.axes_grid1.inset_locator import BboxPatch, BboxConnector,\
     BboxConnectorPatch
fig, ax = plt.subplots(1, 2)
fig.set_size_inches(12, 6)
fig.set_tight_layout(True)
ax[0].set_xlim(0, 1)
ax[0].set_ylim(0, 1)
ax[1].set_xlim(0, 1)
ax[1].set_ylim(0.35, 0.65)
trans1 = blended_transform_factory(ax[0].transData, ax[0].transData)
trans2 = blended_transform_factory(ax[1].transData, ax[1].transData)
bbox = Bbox.from_extents(0, 0.4, 1, 0.6)
bbox1 = TransformedBbox(bbox, trans1)
bbox2 = TransformedBbox(bbox, trans2)
p = BboxConnectorPatch(bbox1, bbox2,
                       loc1a=1, loc2a=2, loc1b=4, loc2b=3, ec="b", fc="b")
p.set_clip_on(False)
ax[0].add_patch(p)
ax[1].yaxis.tick_right()
plt.show()
```

**Actual outcome**

![download](https://cloud.githubusercontent.com/assets/10883104/22823348/77d3aa2c-ef83-11e6-81b2-6c226dac64f4.png)

**Expected outcome**

I would expect the trapezoid to be filled in blue rather than to be empty.

**Matplotlib version**
Matplotlib version: 2.0.0
Python version: 3.4.2
Platform: Linux
Python was installed from the debian repositories and matplotlib was installed via pip.

**Comment by tacaswell** [comment]
Did this use to work with 1.5?

**Comment by tacaswell** [comment]
I confirmed that this is a regression from 1.5

**Comment by tacaswell** [comment]
Something is setting the alpha on the face color to 0 when you go through `set_facecolor`, but if you reach in and set `p._facecolor` directly it renders correctly.

This is....odd.

**Comment by jotasi** [comment]
I can confirm as well that this is a regression from 1.5. I also noted another thing. The trapezoid is shifted a pixel to the left:

Matplotlib 1.5:
![matplotlib1 5](https://cloud.githubusercontent.com/assets/10883104/22832324/ddf27e08-efad-11e6-991b-445ff83a44d0.png)

Matplotlib 2.0:
![matplotlib2 0](https://cloud.githubusercontent.com/assets/10883104/22832409/354873a6-efae-11e6-8400-03c124ee6a8b.png)

Is this just me using a wrong input for the boundaries and the difference is just in the line thickness or is this another small bug? If so, should I open another issue for this or should I just leave that here?

**Comment by tacaswell** [comment]
Ah this makes sense now, the init methods of `BboxConnectorPatch` is setting `fill=False` to fix https://github.com/matplotlib/matplotlib/issues/5757 

There just needs to be some logic in `inset_locator.py` to identify when a face color is passed in and not pass `fill=False`.

In the mean time, the work around is to do

```python
p.set_fill(True)
p.set_facecolor('b')
```

**Comment by jotasi** [comment]
Sorry clicked the wrong button, I did not mean to close this...

**Comment by jotasi** [comment]
Good to know. Thanks @tacaswell!

**Comment by tacaswell** [comment]
And the pixel overflow is likely due to pixel snapping / anti-aliasing / edgewidth / dpi changes.

**Comment by jotasi** [comment]
Ok. Looking at a zoom of the original picture I posted, it seems that the trapezoid's borders are to the left of both axes but that might just be because the line thickness is exactly a pixel in that case.

**Comment by dstansby** [comment]
If anyone is interested in fixing this, https://github.com/matplotlib/matplotlib/pull/8075#issuecomment-285899514 has a proposed solution.

**Comment by jklymak** [comment]
Closed by #10686
